File: Beginner/C/String/CountAndSay.c
char* countAndSay(int n) {
    char* buf = malloc(sizeof(char) * 4500);
    char helper[4500];
    buf[0] = '1'; buf[1] = '\0';
    int size = 1;
    for (int i = 1; i < n; i ++) {
        int pos = 0;
        for (int j = 0; j < size; j ++) {
            int count = 1;
            while (buf[j] == buf[j + 1]) {
                count ++; j ++;
            }
            helper[pos ++] = (count ++) + '0';
            helper[pos ++] = buf[j];
        }
        size = pos;
        for (int k = 0; k < size; k ++)
            buf[k] = helper[k];
        buf[size] = '\0';
    }
    return buf;
}

File: Beginner/Python/String/CountAndSay.py
class Solution(object):
    def countAndSay(self, n):
        """
            :type n: int
            :rtype: str
            """
        buf = ["1"]
        for _ in range(n - 1):
            L = []
            i = 0
            while i < len(buf):
                count = 1
                while i != len(buf) - 1 and buf[i] == buf[i + 1]:
                    count += 1
                    i += 1
                L.append(str(count))
                L.append(buf[i])
                i += 1
            buf = L
        return ''.join(buf)
